superset/databases/commands/validate.py

```diff
@@ -99,9 +100,12 @@ def run(self) -> None:
         except json.decoder.JSONDecodeError:
             encrypted_extra = {}
 
+        # encode parameters
+        params = encode_parameters(self._properties.get("parameters", {}))
```

Comment:
does this value get saved somewhere or is this just for testing?

Reply:
encode paramaters returns the encode keys in a new dictionary (params) which is used to build the sqlalchemy url a few lines below

Reply:
Hugh, I think the problem is that the validation command doesn't save the database, so the encoding will only apply during validation but not later. You also need to update `DatabaseParametersSchemaMixin`.

It might be simpler to implement the encoding logic inside `build_sqlalchemy_uri`. This way it will work with both the validator command and the APIs, and we can have DB-specific encoding logic, which could be needed in the future.